web/ui_utils.js

```diff
@@ -155,7 +155,7 @@ function scrollIntoView(element, spot, scrollMatches = false) {
  * Helper function to start monitoring the scroll event and converting them into
  * PDF.js friendly one: with scroll debounce and scroll direction.
  */
-function watchScroll(viewAreaElement, callback) {
+function watchScroll(viewAreaElement, callback, abortController) {
```

Comment:
This new parameter should probably be optional, and we can probably shorten things slightly.
```suggestion
function watchScroll(viewAreaElement, callback, signal = null) {
```